pkg/java/pom/parse.go

```diff
@@ -86,6 +88,32 @@ func (p *parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 		return nil, nil, xerrors.Errorf("failed to parse POM: %w", err)
 	}
 
+	var remoteRepositories []string
+	for _, rep := range content.Repositories.Repository {
+		if rep.Releases.Enabled == "false" && rep.Snapshots.Enabled == "false" {
+			continue
+		}
+
+		repoURL, err := url.Parse(rep.URL)
+		if err != nil {
+			log.Logger.Debugf("Unable to parse remote repository url: %s", err)
+			continue
+		}
+
+		for _, server := range p.servers {
+			if rep.ID == server.ID && server.Username != "" && server.Password != "" {
+				repoURL.User = url.UserPassword(server.Username, server.Password)
+				break
+			}
+		}
+
+		log.Logger.Debugf("Adding repository %s: %s", rep.ID, rep.URL)
+		remoteRepositories = append(remoteRepositories, repoURL.String())
+	}
+
+	// Add central maven repository or repositories obtained using `WithRemoteRepos` function.
+	remoteRepositories = append(remoteRepositories, p.remoteRepositories...)
```

Comment:
I'm not sure about this.
Do we need to always add central Maven repository or only when remote repositories are not used.
@renypaul wdyt?

Reply:
@DmitriyLewen If we don't mention maven default repository, mvn command still pulls from maven repository. So I believe we should keep the behavior similar to mvn and keep default maven repository

Reply:
@DmitriyLewen Thanks for refactoring the code. It looks like the expected result is slightly different. I will check and update the test code if needed

Reply:
> So I believe we should keep the behavior similar to mvn and keep default maven repository

But if remote repositories are used, then `maven` will check the artifact from those repositories first, right?

Reply:
> update the test code if needed

Thanks!

Reply:
### Following is the output from `mvn help:effective-settings`

This is based on the command mentioned in https://maven.apache.org/guides/mini/guide-multiple-repositories.html#repository-order
```
root@go~(PASS)$ mvn help:effective-settings
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] --- maven-help-plugin:3.4.0:effective-settings (default-cli) @ standalone-pom ---
Effective user-specific configuration settings:

<?xml version="1.0" encoding="UTF-8"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.1.0 http://maven.apache.org/xsd/settings-1.1.0.xsd">
  <localRepository>/root/testdata/user/repository</localRepository>
  <servers>
    <server>
      <id>user-server</id>
    </server>
    <server>
      <username>test-user</username>
      <password>***</password>
      <id>server-with-credentials</id>
    </server>
    <server>
      <username>test-user-only</username>
      <id>server-with-name-only</id>
    </server>
    <server>
      <id>global-server</id>
    </server>
  </servers>
  <pluginGroups>
    <pluginGroup>org.apache.maven.plugins</pluginGroup>
    <pluginGroup>org.codehaus.mojo</pluginGroup>
  </pluginGroups>
</settings>

root@go~(PASS)$ cat ~/.m2/settings.xml
<?xml version="1.0" encoding="UTF-8"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
    <localRepository>testdata/user/repository</localRepository>

    <servers>
        <server>
            <id>user-server</id>
        </server>
        <server>
            <id>server-with-credentials</id>
            <username>test-user</username>
            <password>test-password</password>
        </server>
        <server>
            <id>server-with-name-only</id>
            <username>test-user-only</username>
        </server>
    </servers>
</settings>
root@go~(PASS)$ cat /usr/share/maven/conf/settings.xml
<?xml version="1.0" encoding="UTF-8"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
    <localRepository>testdata/repository</localRepository>

    <servers>
        <server>
            <id>global-server</id>
        </server>
        <server>
            <id>server-with-credentials</id>
            <username>test-user-from-global</username>
            <password>test-password-from-global</password>
        </server>
        <server>
            <id>server-with-name-only</id>
            <username>test-user-only</username>
        </server>
    </servers>
</settings>
```
From the output, the summary is
- mvn looks for server-id only while checking for repository and takes rest of the configuration from the settings.xml where it found first
- mvn give preference to user settings for server id also
@DmitriyLewen Based on the result, I am thinking of updating the code. Please let me know if I am missing something

Reply:
> Based on the result, I am thinking of updating the code. Please let me know if I am missing something

hm... it looks like there are mistakes in the documentation...
User servers > global servers.
It will be great if you update the code.
Can you also add the output of the `mvn help:effective-settings` command to test case (e.g. `hard requirement for the specified version` test) and remove link to docs about servers (to avoid confusion)?

Reply:
Thanks for your work!
I left some nil comments. Take a look, please.